src/test/java/io/reactivex/flowable/FlowableTests.java

```diff
@@ -1060,4 +1060,19 @@ public void toObservableRange() {
     public void toObservableError() {
         Flowable.error(new TestException()).toObservable().test().assertFailure(TestException.class);
     }
+
+    @Test
+    public void zipIterableObject() {
+        final List<Flowable<Integer>> flowables = Arrays.asList(Flowable.just(1, 2, 3), Flowable.just(1, 2, 3));
+        Flowable.zip(flowables, new Function<Object[], Object>() {
```

Comment:
Yup `Function<Integer[], Object>` didn't work for me so I didn't add a test case.